server/src/main/java/org/opensearch/cluster/routing/WeightedRouting.java

```diff
@@ -54,7 +54,9 @@ public boolean isSet() {
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
-        out.writeString(attributeName);
+        if (attributeName != null) {
```

Comment:
This null check can break the read/write from Stream, as during write we will not write null value but during read we still expect attributeName . Please fix it.

Reply:
makes sense to avoid null check. It will hide the issue but still fail the cluster manager node to come up unhealthy in case of bad values.